Some minor fixes for LaTeX printer

oscargus opened the pull request:
<!-- Your title above should be a short description of what
was changed. Do not include the issue number in the title. -->

#### References to other Issues or PRs
<!-- If this pull request fixes an issue, write "Fixes #NNNN" in that exact
format, e.g. "Fixes #1234". See
https://github.com/blog/1506-closing-issues-via-pull-requests . Please also
write a comment on that issue linking back to this pull request once it is
open. -->
Closes #15934

#### Brief description of what is fixed or changed
Some fixes for the LaTeX printer:
 * Always use `\text` instead of `\mathrm` (see #15934 )
 * Use `mathbf` instead of `\boldmath{\mathrm` (gives identical rendering)
 * Added tests for some DiffGeom printing
 * Made a unified function for printing of transforms
 * Corrected documentation for `imaginary_unit`

#### Other comments
`\text` is from `amsmath` but we rely on that heavily elsewhere. The "problem" with `\mathrm` is that `rm` stand for a Roman (Serif) font. In certain situations, the template use non-Roman fonts for the math and then `\text` works better as it simply gives the non-italic version of the math font.

#### Release Notes

<!-- Write the release notes for this release below. See
https://github.com/sympy/sympy/wiki/Writing-Release-Notes for more information
on how to write release notes. The bot will check your release notes
automatically to see if they are formatted correctly. -->

<!-- BEGIN RELEASE NOTES -->
* printing
  * `\text` is consistently used instead of `\mathrm` in LaTeX printer to get non-italic text.
  * Added option to select output format for `re` and `im` in LaTeX printer: `gothic_re_im` 
<!-- END RELEASE NOTES -->

oscargus committed:
Made a single printing function for transforms

oscargus committed:
Changed '\boldmath{\mathrm{...}}' to '\mathbf{...}'

oscargus committed:
Changed mathrm to text and added tests

oscargus commented on sympy/printing/tests/test_latex.py:
```diff
@@ -1958,3 +1958,16 @@ def test_imaginary_unit():
+
+def test_DiffGeomMethods():
+    from sympy.diffgeom import Manifold, Patch, CoordSystem, Point, BaseScalarField
+    r0, theta0 = symbols('r0, theta0')
```
These were apparently used later on in the code snippet I copied from... Will remove them, but waiting for review comments first.

oscargus committed:
Fixed sloppy error and used format instead of %

asmeurer commented on sympy/printing/latex.py:
```diff
@@ -1494,7 +1494,7 @@ def _print_Transpose(self, expr):
     def _print_Trace(self, expr):
         mat = expr.arg
-        return r"\mathrm{tr}\left(%s \right)" % self._print(mat)
+        return r"\text{tr}\left(%s \right)" % self._print(mat)
```
This should be \operatorname

oscargus replied:
Makes sense.

asmeurer commented:
Are all the modified lines tested?

oscargus commented:
For the `format` ones I am not sure as the output doesn't change there I have not checked it up.

Looking at the `text` ones there are a few which doesn't seem to have a test. I added a few which I noted didn't have a test but other than that I more or less blindly replace `mathrm` with `text` in the tests. So didn't really reflect on it. It seems like `Differential` does not have a test currently.

oscargus committed:
Fixed review comments

oscargus committed:
Fixed test

oscargus commented:
I've added a test for `Differential` (it turned out that is was tested as part of the `WedgeProduct` test, but not explicitly earlier.

For the other methods I have changed, I believe most are tested although again not explicitly created from that class. These are all Modules from the `polys.agca` part and there are basically no examples so I do not really know how to create them...

oscargus committed:
File ending...

oscargus committed:
Fixed LaTeX printing of im and added configurability

oscargus commented:
I've made printing of `im`consistent with printing of `re`.

I also added an option to change the output between `\Re` and `\operatorname{re}` called `text_re_im`, which is `False` by default leading to the former output (which is what Sympy has used historically. @jksuom do you have opinions on the flag name?

asmeurer commented:
How about `re_im_style` and it's a string instead of a boolean. text_re_im is misleading because it uses \operatorname, not \text. 

By the way, I vote to change the re/im style to by default be \operatorname. As I mentioned [here](https://github.com/sympy/sympy/pull/16065#issuecomment-468852020), the gothic style looks terrible, and it's rarely used as @jksuom noted (apparently it [originates from Knuth himself](https://tex.stackexchange.com/questions/192692/why-does-amsmath-use-fraktur-for-real-and-imaginary-parts)).

oscargus commented:
What about `gothic_re_im` with Boolean? If ext-based, what are the options called? (I originally considered this but couldn't figure out the classifiers.) I agree that the current one may be a bit confusing, but the end result is text, although not `\text`...

jksuom commented:
> I vote to change the re/im style to by default be \operatorname.

+1 

As to the keyword, `gothic_re_im` looks good to me. (There will probably be no need to use it explicitly if its default is false.)

oscargus committed:
Changed text_re_im to gothic_re_im

oscargus commented:
Changed to `gothic_re_im` with default `False`.

oscargus committed:
Fixed stupid spelling error

oscargus committed:
Fixed even more stupid spelling error

jksuom commented on sympy/printing/latex.py:
```diff
@@ -2376,7 +2376,7 @@ def latex(expr, fold_frac_powers=False, fold_func_brackets=False,
     gothic_re_im : boolean, optional
         If set to ``True``, `\Re` and `\Im` is used for ``re`` and ``im``, respectively.
-        The default is ``False`` leading to `\opeatorname{re}` and `\opeatorname{re}`.
+        The default is ``False`` leading to `\operatorname{re}` and `\operatorname{re}`.
```
`re` and `im`?

oscargus replied:
Yes, that is the next commit two minutes later...

oscargus committed:
Merge branch 'master' into latexprinterfixes

oscargus committed:
Fixed printing of RandomDomain

oscargus committed:
Better test coverage

oscargus committed:
Merge with master, resolved conflicts, shortened lines

oscargus commented:
This is fixed and merged with current master.
